mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -222,134 +280,423 @@ public void testSimpleAttachment3() throws MessagingException, IOException {
                 .mailetName("Test")
-                .build());
+                .setProperty("directory", folderPath)
+                .setProperty("remove", "matched")
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.removeAttachments).isEqualTo("matched");
+    }
 
+    @Test
+    public void serviceShouldThrowWhenUnretrievableMessage() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+        
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenThrow(new MessagingException("Test exception"));
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not retrieve message from Mail object");
+        
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenUnretrievableContentTypeMessage() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+
+        MimeMessage message = mock(MimeMessage.class);
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenReturn(message);
+        when(message.isMimeType("multipart/*"))
+            .thenThrow(new MessagingException("Test exception"));
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not retrieve contenttype of message.");
+        
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenNotAnalyzableMessage() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+
+        MimeMessage message = mock(MimeMessage.class);
+        when(message.isMimeType("multipart/*"))
+            .thenThrow(new RuntimeException("Test exception"));
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenReturn(message);
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not analyse message.");
+
+        mailet.service(mail);
+    }
+
+    @Test
+    public void getMailetInfoShouldReturn() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+
+        assertThat(mailet.getMailetInfo()).isEqualTo("StripAttachment");
+    }
+
+    @Test
+    public void analyseMultipartPartMessageShouldReturnFalseWhenPartIsNotMultipart() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        Mail mail = mock(Mail.class);
+        //When
+        boolean actual = mailet.analyseMultipartPartMessage(part, mail);
+        //Then
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void analyseMultipartPartMessageShouldReturnTrueWhenAtLeastOneMultipartShouldHaveBeenRemoved() throws Exception {
```

Comment:
If so analyze do more than just analyzing and actually take decisions.

 - Split it in subMethods
 - Or rename it.